### ISSUE TITLE
[apex] Database.query not showing violation for ApexCRUDViolation

### ISSUE THREAD

**Issue by praful-gupta** [pmd/pmd#5792]
**Affects PMD Version:** Latest 

**Rule:** [ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**
ApexCRUDViolation is not thrown if there is no permission check happening, as below:

```
 string  query = 'Select Id, Name, xyz__c FROM Account  limit 10';
 List<Account> accountList = Database.query(query);
```

As per the Salesforce documentation, it's possible to define the accessLevel as AccessLevel.USER_MODE instead of running in System_mode as default.
https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_database.htm

**Expected outcome:**

PMD should report a violation at line 2, but doesn't as the SOQL is running in System mode without permission checks. This is a false-negative.

**Running PMD through:** *[CLI]*

It seems there was a similar issue in past, but it wasn't resolved above use case.

**Comment by adangel** [comment]
Thanks for the report. This is actually a duplicate of #2628.
